Assets/Scripts/Enemy/EnemyController.cs

```diff
@@ -3,47 +3,57 @@
 using UnityEngine;
 using UnityEngine.AI;
 
-public class EnemyController : MonoBehaviour {
+public class EnemyController : EntityController {
 
     [SerializeField]
     private Weapon weapon;
     [SerializeField]
     private float hitChance;
 
     private GameObject target;
+    private EntityController targetController;
     private NavMeshAgent agent;
     private bool targetInRange = false;
     private float deltaTime;
 
-	void Start () {
+	protected override void Start () {
+        base.Start ();
         this.target = GameObject.FindGameObjectWithTag ("Player");
+        this.targetController = target.GetComponent<PlayerController> ();
         this.agent = GetComponent<NavMeshAgent> ();
+        this.totalHealth = 30f;
+        this.health = totalHealth;
 	}
 	
-	void Update () {
+	protected override void Update () {
+        base.Update ();
```

Comment:
Yeah, I'm fine with doing it this way. I don't understand why it's better, but I'll 100% trust your experience in the matter.